lib/Controller/LoginController.php

```diff
@@ -260,11 +262,15 @@ public function login(int $providerId, string $redirectUrl = null) {
 			}
 		}
 
+		$redirectUrl = $this->urlGenerator->linkToRouteAbsolute(Application::APP_ID . '.login.code');
+		if ($externalSslTermination) {
+			$redirectUrl = preg_replace('/^http:\/\//i', 'https://', $redirectUrl);
```

Comment:
This should already be possible to achieve by setting proper reverse proxy headers or or the overwrite protocol config option:

https://docs.nextcloud.com/server/latest/admin_manual/configuration_server/reverse_proxy_configuration.html

I don't think we need an extra setting in user_oidc for that

Reply:
Very true. I thought this could still be useful in some cases when admins don't have control over the reverse proxy config.

Reply:
For that purpose they can set the overwrite protocol ;)

Reply:
Very true again...sorry for the noise.

Reply:
No worries :)